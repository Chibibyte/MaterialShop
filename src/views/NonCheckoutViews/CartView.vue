<template>
  <v-container v-if="this.$store.state.noncheckout.cart.length > 0">
    <v-row>
      <v-col>
        <v-row v-for="item in cart" :key="item.id">
          <ItemListViewItem
            :item="items[item.id]"
            :initAmount="item.amount"
            clearingCart
            updateOnChange
          />
        </v-row>
      </v-col>
      <v-col cols="3">
        <v-card class="white d-flex justify-center align-items-center">
          <div>
            <v-card-title
              >Sum ({{ this.$store.getters["noncheckout/amount"] }} Artikel):
              {{ this.$store.getters["noncheckout/price"] }}â‚¬</v-card-title
            >
            <v-card-actions>
              <CheckoutButton />
            </v-card-actions>
          </div>
        </v-card>
      </v-col>
    </v-row>
  </v-container>
  <v-container v-else>
    <div class="text-center text-h1">Your cart is empty</div>
  </v-container>
</template>

<script>
import CheckoutButton from "@/components/CheckoutButton.vue";
import ItemListViewItem from "@/components/ItemListViewItem.vue";
export default {
  name: "CartView",
  components: {
    ItemListViewItem,
    CheckoutButton,
  },
  computed: {
    items() {
      return this.$store.getters["noncheckout/items"];
    },
    cart() {
      return this.$store.state.noncheckout.cart;
    },
  },
};
</script>