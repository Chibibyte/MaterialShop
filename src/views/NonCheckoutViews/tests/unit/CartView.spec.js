
import Vuex from 'vuex'
import { shallowMount, createLocalVue } from '@vue/test-utils';
import CartView from '@/views/NonCheckoutViews/CartView.vue';

const localVue = createLocalVue();
localVue.use(Vuex);

let store;
let state;
let modules;
let stubs = [
    'v-container',
    'v-row',
    'v-col',
    'v-card',
    'v-card-title',
    'v-card-actions'
]
let wrapper;

beforeEach(() => {
    state = {}
    modules = {
        noncheckout: {
            namespaced: true,
            state() {
                return {
                    cart: [
                        {
                            id: 3,
                            amount: 4
                        },
                        {
                            id: 4,
                            amount: 2
                        }
                    ]
                }
            },
            getters: {
                amount: jest.fn(),
                price: jest.fn(),
                items: jest.fn((id) => { return { title: 'a', price: 2.99, id, amount: 3 } })
            }
        }
    }
    store = new Vuex.Store({
        state,
        modules
    })
    wrapper = shallowMount(CartView, { stubs, store, localVue });
})

// initial
describe('Component', () => {
    test('is instantiated', () => {
        expect(wrapper.isVueInstance).toBeTruthy();
    });
    test('default snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
});

// computed
describe('computed', () => {
    test('items', () => {
        expect(modules.noncheckout.getters.amount.mock.calls).toHaveLength(1);
    })

    test('cart', () => {
        expect(wrapper.vm.cart).toEqual(wrapper.vm.$store.state.noncheckout.cart);
    })
})