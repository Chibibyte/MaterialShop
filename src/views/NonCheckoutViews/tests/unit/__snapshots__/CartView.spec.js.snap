// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component default snapshot 1`] = `
<v-container-stub>
  <v-row-stub>
    <v-col-stub>
      <v-row-stub>
        <itemlistviewitem-stub
          clearingcart="true"
          initamount="4"
          item="[object Object]"
        />
      </v-row-stub>
      <v-row-stub>
        <itemlistviewitem-stub
          clearingcart="true"
          initamount="2"
          item="[object Object]"
        />
      </v-row-stub>
    </v-col-stub>
     
    <v-col-stub
      cols="3"
    >
      <v-card-stub
        class="white d-flex justify-center align-items-center"
      >
        <div>
          <v-card-title-stub>
            Sum ( Artikel): â‚¬
          </v-card-title-stub>
           
          <v-card-actions-stub>
            <navbutton-stub
              icon="mdi-cash-register"
              name="Checkout"
              to="/checkout"
            />
          </v-card-actions-stub>
        </div>
      </v-card-stub>
    </v-col-stub>
  </v-row-stub>
</v-container-stub>
`;
